### PR TITLE
Update boilerplate template - Advanced Node & Express

### PR DESCRIPTION
**Pull request by jeremyfiel**
Hello

I updated the boilerplate to remove a bunch of local files originally committed which didn't pertain to the FCC curriculum.  I also updated the readme, package.json and removed any references to Hyperdev in favor of Glitch. 
I'm also attempting to move back to the master branch because the "gomix" branch is not customary, especially for new developers.

Let me know if anything is missing. Thanks a lot!

### CODE REVIEW

**Comment by scissorsneedfoodtoo** [comment]
@jeremyfiel, thanks for taking the time to submit this PR. You're right that the `gomix` branch is not customary, but it's currently what we use to sync the boilerplates between GitHub and Glitch (formally Gomix, and Hyperdev before that).

Since this PR mainly brings things from the `gomix` branch into the `master` branch, I think we can safely close this and focus on your other PR: https://github.com/freeCodeCamp/freeCodeCamp/pull/38572

**Comment by jeremyfiel** [comment]
> @jeremyfiel, thanks for taking the time to submit this PR. You're right that the `gomix` branch is not customary, but it's currently what we use to sync the boilerplates between GitHub and Glitch (formally Gomix, and Hyperdev before that).
> 
> Since this PR mainly brings things from the `gomix` branch into the `master` branch, I think we can safely close this and focus on your other PR: [freeCodeCamp/freeCodeCamp#38572](https://github.com/freeCodeCamp/freeCodeCamp/pull/38572)

Not only does this PR move everything to the master branch, I updated all of the deps and removed a bunch of unnecessary local files which were included previously.  Also removed references to hyperdev. I changed the fcc base to merge into gomix so that doesn't break but the rest of the changes shouldn't be affected.  Please review again and let me know if you can merge.   thanks for your help.

**Comment by ojeytonwilliams** [comment]
> but it's currently what we use to sync the boilerplates between GitHub and Glitch 

Can't we just change the default branch from `gomix` to `master`?  That should mean the Glitch clone link uses `master`.  Apologies if I'm missing something obvious!

**Comment by jeremyfiel** [comment]
> > but it's currently what we use to sync the boilerplates between GitHub and Glitch
> 
> Can't we just change the default branch from `gomix` to `master`? That should mean the Glitch clone link uses `master`. Apologies if I'm missing something obvious!

that was my intention.  Let me know if I should change it back to master.

File: server.js
Comment (ojeytonwilliams) [inline server.js]:
```suggestion
  res.sendFile(process.cwd() + "/views/index.html");
```
Changing this from `sendFile` to `render` is part of the challenge, so we should leave it for the user to do.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Thanks for the clean up!  Just the one issue:

File: server.js
Comment (jeremyfiel) [reply server.js]: related conversation on the approach for the template. 
https://github.com/freeCodeCamp/freeCodeCamp/pull/38572#issuecomment-616644655

**Comment by jeremyfiel** [comment]
I must've clicked the wrong thing for your requested changes.  Now a review is pending.   Anyway, I committed the changes in {1f12bf05c7e6669fe4f784c1cf63fc8ee077bf12} to just have text instruction for the boilerplate.  I also updated the instructions and added a new test to the challenge {freeCodeCamp/freeCodeCamp/pull/38572}.   Let me know if you want to change the github base back to master and then you can set the default accordingly.

**Comment by scissorsneedfoodtoo** [comment]
Hi @jeremyfiel and @ojeytonwilliams, thanks for carrying on the discussion here, too.

@ojeytonwilliams, that's a good point -- I'm not sure if Glitch looks at the repos default branch or not before pulling changes over. It would be nice to switch everything over to the more customary `master` branch like @jeremyfiel mentioned. 

Your cleanup looks good so far. The only thing I'm hesitant to remove is `.hyperdev-assets`. I created a new Express project on Glitch and exported it as a .zip file. It seems like `.hyperdev-assets` is now called `.glitch-assets`. I'm not sure exactly what it's used for, though.

Rather than removing it entirely, perhaps we could just rename it to `.glitch-assets` for now.

**Comment by jeremyfiel** [comment]
> Hi @jeremyfiel and @ojeytonwilliams, thanks for carrying on the discussion here, too.
> 
> @ojeytonwilliams, that's a good point -- I'm not sure if Glitch looks at the repos default branch or not before pulling changes over. It would be nice to switch everything over to the more customary `master` branch like @jeremyfiel mentioned.

according to the url they use, it's clone-from-repo and the github URL which points to "default", if it's not specified. 
https://glitch.com/edit/#!/remix/clone-from-repo?REPO_URL=https://github.com/freeCodeCamp/boilerplate-advancednode/
> 
> Your cleanup looks good so far. The only thing I'm hesitant to remove is `.hyperdev-assets`. I created a new Express project on Glitch and exported it as a .zip file. It seems like `.hyperdev-assets` is now called `.glitch-assets`. I'm not sure exactly what it's used for, though.
> 
> Rather than removing it entirely, perhaps we could just rename it to `.glitch-assets` for now.

These assets are generated by glitch for each exported project.  They are the images used on their site like the fish logo and the backgrounds.  They are not necessary for the boilerplate to host.

**Comment by ojeytonwilliams** [comment]
> These assets are generated by glitch for each exported project. They are the images used on their site like the fish logo and the backgrounds. They are not necessary for the boilerplate to host.

Yep, if you create a project from a repo that does not have `.glitch-assets` then that file will be created when you zip up the project.

@jeremyfiel could you remove the merge conflicts?  By the way, it should not be necessary to merge locally unless someone has modified code on the remote.

**Comment by jeremyfiel** [comment]
@ojeytonwilliams  Ok. I think we are set to go now. I removed the merge commit and fixed the conflicts.  Don't forget to check if we should change to the master base and set the default.

File: server.js
Comment (ojeytonwilliams) [inline server.js]:
```suggestion
  //Write your code here to render the Pug template
  res.send('pug template should be rendered instead of this');
```
I still wonder if this would be better than having the site hang until the user fixes it, but I'll defer to @scissorsneedfoodtoo on this one.

**Review by ojeytonwilliams — APPROVED** [review]
LGTM, reservations about the site hanging notwithstanding.

re: `master` vs `gomix` we can just test it to make sure.  Merge this into `master`, change the default branch from `gomix` to `master` and immediately try to clone the repo using https://glitch.com/edit/#!/remix/clone-from-repo?REPO_URL=https://github.com/freeCodeCamp/boilerplate-advancednode/ and see which one we get.

File: server.js
Comment (jeremyfiel) [reply server.js]: ok. i added that response.  Makes sense to show something in the response. I changed to master. If you are going to merge, make sure you set it as default.  I don't have that access.

**Comment by ojeytonwilliams** [comment]
@jeremyfiel let me clean up the history a bit.

**Comment by ojeytonwilliams** [comment]
Actually, I just realised this is on your `master` branch, so I cannot help.  So, in order for me to be able to help you can:

```
git checkout master
git checkout -b fix/update-boilerpate
git push -u origin HEAD
```

At this point you can create a new PR with that branch and I can clean things up.

**Comment by jeremyfiel** [comment]
> Actually, I just realised this is on your `master` branch, so I cannot help. So, in order for me to be able to help you can:
> 
> ```
> git checkout master
> git checkout -b fix/update-boilerpate
> git push -u origin HEAD
> ```
> 
> At this point you can create a new PR with that branch and I can clean things up.

i fixed her up.

**Comment by ojeytonwilliams** [comment]
@jeremyfiel Nice one! @scissorsneedfoodtoo I think this is good to go, depending on what you think about what I said regarding testing `master`.

**Comment by scissorsneedfoodtoo** [comment]
Awesome! Thanks both of you for all of your hard work on this. 

@ojeytonwilliams, sure, that sounds like a plan. I don't seem to have access to the settings for the repo now. I'll need to look into that real quick before merging and testing.

**Comment by scissorsneedfoodtoo** [comment]
@jeremyfiel and @ojeytonwilliams, thanks again for updating this boilerplate. Got access to the settings on this repo. Will merge and switch the default branch from `gomix` to `master` now.

**Comment by scissorsneedfoodtoo** [comment]
Tested remixing via https://glitch.com/edit/#!/remix/clone-from-repo?REPO_URL=https://github.com/freeCodeCamp/boilerplate-advancednode/ and the Glitch "Clone from Git Repo" option and everything seems to be working!

**Pull request merged**
